### PR TITLE
[CPU]Fix qwen rope fusion with special_zero

### PR DESCRIPTION
**Pull request by zhangYiIntel**
### Details:
 - *Extend Qwen Rope Fusion*
 - PR to master branch https://github.com/openvinotoolkit/openvino/pull/24750

### Tickets:
 - *CVS-142523*

### CODE REVIEW

**Comment by dmitry-gorokhov** [comment]
@v-Golubev Please take a look. Thanks!

File: src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/rope_fusion.cpp [hunk 1]
```diff
@@ -746,6 +746,34 @@ ov::intel_cpu::RoPEFusionQwen::RoPEFusionQwen(int split_output_id) {
...
+            auto check_const = [&data_shape](Output<Node> output) -> bool {
+                auto node = std::dynamic_pointer_cast<opset1::Constant>(output.get_node_shared_ptr());
+                const auto& target = node->cast_vector<int32_t>();
+                // ensure data_shape and target_shape have correct rank
+                if (target.size() < 3 || data_shape.rank().is_dynamic() || (data_shape.rank().is_static() && data_shape.rank().get_length() < 2)) {
+                    return false;
+                }
+                if (data_shape[data_shape.size() - 1].is_static() && data_shape[data_shape.size() - 2].is_static()) {
+                    auto head_size = data_shape[data_shape.size() - 1].get_length();
+                    auto head_cnt = data_shape[data_shape.size() - 2].get_length();
```
Comment (v-Golubev) [inline src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/rope_fusion.cpp:763]: `RoPENode::Config` [already has](https://github.com/openvinotoolkit/openvino/pull/24727/files#diff-8c8905f6745d31f66e0f0dfae7c5b5d614fd03c3b643f0f29250f24879b2a9f4R735-R736) the filled `head_cnt` and `head_size` values. Maybe we can reuse them in this reshape check?

File: src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/rope_fusion.cpp [hunk 2]
```diff
@@ -746,6 +746,34 @@ ov::intel_cpu::RoPEFusionQwen::RoPEFusionQwen(int split_output_id) {
             config.slice_stop = config.slice_start + config.head_cnt * config.head_size;
         }
 
+        if (pattern_map.count(reshape_special)) {
+            // check reshape_special shape correctness
+            auto reshape_special_node = pattern_map.at(reshape_special).get_node_shared_ptr();
+            auto data_shape = reshape_special_node->get_input_partial_shape(0);
+            auto reshape_shape = pattern_map.at(const_shape);
+            auto check_const = [&data_shape](Output<Node> output) -> bool {
+                auto node = std::dynamic_pointer_cast<opset1::Constant>(output.get_node_shared_ptr());
```
Comment (v-Golubev) [inline src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/rope_fusion.cpp:755]: It's better to use OV RTTI for casting:
```suggestion
                auto node = ov::as_type_ptr<opset1::Constant>(output.get_node_shared_ptr());
```

File: src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/rope_fusion.cpp [hunk 3]
```diff
@@ -746,6 +746,34 @@ ov::intel_cpu::RoPEFusionQwen::RoPEFusionQwen(int split_output_id) {
             config.slice_stop = config.slice_start + config.head_cnt * config.head_size;
         }
 
+        if (pattern_map.count(reshape_special)) {
+            // check reshape_special shape correctness
+            auto reshape_special_node = pattern_map.at(reshape_special).get_node_shared_ptr();
+            auto data_shape = reshape_special_node->get_input_partial_shape(0);
+            auto reshape_shape = pattern_map.at(const_shape);
+            auto check_const = [&data_shape](Output<Node> output) -> bool {
```
Comment (v-Golubev) [inline src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/rope_fusion.cpp:754]: Do we really need this lambda? It is used only once, so we could just use its body and avoid lambda creation

File: src/plugins/intel_cpu/tests/functional/custom/subgraph_tests/src/rotary_pos_emb.cpp [hunk 4]
```diff
@@ -325,8 +325,15 @@ TEST_F(RoPECPUTestChatGLM, smoke_CompareWithRefs) {
     CheckNumberOfNodesWithType(compiledModel, "RoPE", 1);
 }
 
-class RoPECPUTestQwen7b : public SubgraphBaseTest {
+class RoPECPUTestQwen7b : public SubgraphBaseTest, public testing::WithParamInterface<bool> {
 public:
+    static std::string getTestCaseName(const testing::TestParamInfo<bool>& obj) {
+        bool specialReshape;
+        specialReshape = obj.param;
```
Comment (v-Golubev) [inline src/plugins/intel_cpu/tests/functional/custom/subgraph_tests/src/rotary_pos_emb.cpp:332]:
```suggestion
        const bool specialReshape = obj.param;
```

**Review by v-Golubev — APPROVED** [review]
I have only minor suggestions which don't affect the logic. I don't think that they block the merge: we can apply them in follow-up PR to master branch

File: src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/rope_fusion.cpp (hunk 3 above)
Comment (zhangYiIntel) [reply src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/rope_fusion.cpp:754]: It could be removed

**Review by usstq — APPROVED** [review]
LGTM

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**